### DISCUSSION TITLE
How to change HTML error page ?

### DISCUSSION THREAD

**Discussion by LM1LC3N7** [nginx-proxy/nginx-proxy#1956] [Q&A]
Hello,

I would like to change HTML error page, and I try to add to `my_proxy.conf` :

``` sh
if (-f $document_root/error503.html) {
    return 503;
}

error_page 503 @maintenance;

location @maintenance {
    rewrite ^(.*)$ /error503.html break;
}
```

Logs :

``` sh
2016/03/23 18:33:35 [emerg] 17#17: "if" directive is not allowed here in /etc/nginx/conf.d/my_proxy.conf:3
```

So I remove `if` :

``` sh
2016/03/23 18:27:37 [emerg] 17#17: "location" directive is not allowed here in /etc/nginx/conf.d/my_proxy.conf:3
```

When I launch nginx, I add this line :

``` sh
 -v /home/reverse_proxy/html:/usr/share/nginx/html 
```

But it seems to doesn't work !

Do you know how I can do that ?

**Comment by bhurlow** [comment]
looks like the template returns a generic 503 when there are no vhosts: https://github.com/jwilder/nginx-proxy/blob/master/nginx.tmpl#L54-L59

**Comment by LM1LC3N7** [comment]
I don't understand how to do to force nginx to use my 503 (and 404) error page.

In vhost.d folder, their is only one file : `default`

``` sh
## Start of configuration add by letsencrypt container
location /.well-known/ {
    auth_basic off;
    root /usr/share/nginx/html;
    try_files $uri =404;
}
## End of configuration add by letsencrypt container

```

**Comment by bhurlow** [comment]
one option would be something like this:

```
server {
    server_name _; # This is just an invalid value which will never trigger on a real hostname.
    listen 80;
    access_log /var/log/nginx/access.log vhost;
    error_page 404 /50x.html;
    location = /50x.html {
        root /usr/share/nginx/html;
    }
}
```

however, this would still return a `404` along with the error html page

**Comment by LM1LC3N7** [comment]
so, I add

``` sh
server {
    server_name _; # This is just an invalid value which will never trigger on a real hostname.
    listen 80;
    access_log /var/log/nginx/access.log vhost;
    error_page 404 /50x.html;
    location = /50x.html {
        root /usr/share/nginx/html;
    }
}
```

To my default file, or to `my_proxy.conf` ?

**Comment by dm17** [comment]
> looks like the template returns a generic 503 when there are no vhosts: https://github.com/jwilder/nginx-proxy/blob/master/nginx.tmpl#L54-L59

Is there a way to have custom error messages when there are no vhosts?

**Comment by 1-alex98** [comment]
Looking for the same... How do I do it???
I tried adding an error page for 503 but if my service is down the server{...} stuff gets removed and my error page with it

**Comment by dm17** [comment]
> Looking for the same... How do I do it???
> I tried adding an error page for 503 but if my service is down the server{...} stuff gets removed and my error page with it

I switched to traefik.

**Comment by 1-alex98** [comment]
I mean I could add:
http{
...
    error_page 503 /503.html;
...
}
but then where do I put it... sadly I can not use location in the HTTP directive

**Comment by 1-alex98** [comment]
> > Looking for the same... How do I do it???
> > I tried adding an error page for 503 but if my service is down the server{...} stuff gets removed and my error page with it
> 
> I switched to traefik.

Well not sure if that's what I want but yes if I would do it again I would probably also choose traffic

**Comment by 1-alex98** [comment]
I mean everything else worked so far just the error page is a problem :thinking:

**Comment by 1-alex98** [comment]
oh ok jetzt verstehe ich wo die 503 her kommt https://github.com/nginx-proxy/nginx-proxy/blob/c8a6785898e1ac860f5734a1eac6d395158a777f/nginx.tmpl#L151

**Comment by dm17** [comment]
> > > Looking for the same... How do I do it???
> > > I tried adding an error page for 503 but if my service is down the server{...} stuff gets removed and my error page with it
> > 
> > 
> > I switched to traefik.
> 
> Well not sure if that's what I want but yes if I would do it again I would probably also choose traffic

Caddy is another option. Traefik is quite opaque in some ways, but nginx takes the cake...

**Comment by 1-alex98** [comment]
![Screenshot from 2020-12-12 23-21-29](https://user-images.githubusercontent.com/19996713/101996238-ccb1ca80-3cd0-11eb-8a0c-076a3f1f0904.png)
![Screenshot from 2020-12-12 23-20-59](https://user-images.githubusercontent.com/19996713/101996239-cd4a6100-3cd0-11eb-91fd-25dcbbc13ea3.png)

**Comment by 1-alex98** [comment]
The problem is even tho now it returns the right error page the cert is wrong

**Comment by 1-alex98** [comment]
bad setup for an error page

**Comment by 1-alex98** [comment]
@dm17 you might be right that traffic is probably more suitable

**Comment by dm17** [comment]
Would help but I'd have to try to remember nginx... If you followed the tutorial, then I guess you saw that it is unlikely that you have to edit those config files yourself (many of them are auto-generated).
